### DISCUSSION TITLE
Is there a way to build a custom EmojiPopupComponent for @remirror/react?

### DISCUSSION THREAD

**Discussion by JanReichelt** [remirror/remirror#2057] [Q&A]
I am currently working with the [Remirror-Editor](https://remirror.io/) - the emoji extension is integrated and works fine. The EmojiPopupComponent from `@remirror/react` works fine as well, but i would like to refine it for project specific needs.

Is there a way to extend this component or to write a custom EmojiPopupComponent? If so, what does it need to get it running?

I skimmed the (Remirror)-Documentation, tried to find the actual EmojiPopupComponent in my node_modules and the Remirror github, but wasn't able to wrap my head around this issue.

```
import { EmojiExtension } from 'remirror/extensions'
import {
  EmojiPopupComponent,
  Remirror,
  useRemirror,
} from '@remirror/react'

const Editor = () => {
  const extensions = useCallback(
    () => [
      /// ... other Extensions...
      new EmojiExtension({
        data: emojiData,
        moji: 'noto',
        fallback: ':smile:',
        suggestionCharacter: ':',
        plainText: true,
      }),
    []
  )

  const { manager } = useRemirror({ extensions })

  // ...

  return (
    <Remirror manager={manager}>
      <EmojiPopupComponent />
      {children}
    </Remirror>
  )
}

export default Editor
```

This is a trimmed down example of how i use Remirror and the extension in question. I would like to replace the `<EmojiPopupComponent />` with my own, custom one if possible. But this one doesn't accept props, i guess it uses data from the Remirror context. Which data is necessary to make a custom EmojiPopupComponent functional and how do i access it?

**Comment by c0ntradicti0n** [comment]
This can be customized:

https://github.com/remirror/remirror/blob/5a6ade6e35f5480eca1eac8b281e8a9342409a46/packages/remirror__react-components/src/popups/emoji-popup-component.tsx#L28

Good luck
